src/kOS/Function/Misc.cs

```diff
@@ -322,7 +322,9 @@ public override void Execute(SharedObjects shared)
                 // or to a file to save:
                 if (justCompiling)
                 {
-                    List<CodePart> compileParts = shared.ScriptHandler.Compile(path, 1, fileContent.String, String.Empty, options);
+                    // since we've already read the file content, use the volume from outPath instead of the source path
+                    volume = shared.VolumeMgr.GetVolumeFromPath(outPath);
+                    List<CodePart> compileParts = shared.ScriptHandler.Compile(path, 1, fileContent.String, string.Empty, options);
                     VolumeFile written = volume.SaveFile(outPath, new FileContent(compileParts));
```

Comment:
Agreed.  It probably should mean that `VolumeManager` allows us to save the file instead of the volume itself.  But trying to do that seemed beyond this fix (and apparently you agree since you just merged)